Automattic/wp-calypso
# Site Profiler: add redacted fields check

Fixes: https://github.com/Automattic/wp-calypso/issues/82185

## Proposed Changes

* Add check for redacted fields. The actual check covers the four known placeholder texts
* Fix: the `*_street` WHOIS values can also be an array of strings

## Testing Instructions
* Go to http://calypso.localhost:3000/site-profiler
* Try a site without known redacted fields; nothing should happen
* Try a site with redacted fields like `news.ycombinator.com` or `wordpress.org`. The `Some fields have been redacted for privacy` string should appear and the fields must be hidden and only viewable in the raw WHOIS

![image](https://github.com/Automattic/wp-calypso/assets/167611/990eb36c-7065-4d94-bef9-45f668e7b9fa)



## Pre-merge Checklist
